datumaro/plugins/coco_format/extractor.py

```diff
@@ -124,6 +161,55 @@ def _load_items(self, loader):
 
         return items
 
+    def _load_panoptic_items(self, config, panoptic_images):
+        import numpy as np
+        import PIL.Image
+        from collections import defaultdict
+        items = OrderedDict()
+
+        def rgb2id(color):
+            if isinstance(color, np.ndarray) and len(color.shape) == 3:
+                if color.dtype == np.uint8:
+                    color = color.astype(np.int32)
+                return color[:, :, 0] + 256 * color[:, :, 1] + 256 * 256 * color[:, :, 2]
+            return int(color[0] + 256 * color[1] + 256 * 256 * color[2])
+
+        imgs_info = {}
+        for img in config['images']:
+            imgs_info[img['id']] = img
+
+        for ann in config['annotations']:
+            img_id = ann['image_id']
+            image_path = osp.join(self._images_dir, imgs_info[img_id]['file_name'])
+            image_size = (imgs_info[img_id].get('height'), imgs_info[img_id].get('width'))
+            image = Image(path=image_path, size=image_size)
+            anns = list()
+
+            panoptic_image_path = osp.join(panoptic_images, ann['file_name'])
+            panoptic_format = np.array(PIL.Image.open(panoptic_image_path), dtype=np.uint32)
```

Comment:
Consider using `datumaro.util.mask_tools.load_mask` and `.lazy_mask` (which loads through image cache)

Reply:
I failed with using all that pretty lazy mask functions - they did not what I want